src/Tribe/Main.php

```diff
@@ -2384,24 +2384,37 @@ public function _set_locale() {
 
 		/**
 		 * Redirect the legacy past/upcoming view URLs to list
+         *
+         * @since TBD revised to avoid unwanted redirects
 		 */
 		public function redirect_past_upcoming_view_urls() {
-
-			if ( strpos( $_SERVER['REQUEST_URI'], $this->getRewriteSlug() . '/' . $this->pastSlug ) !== false ) {
-				$search = '#/' . $this->pastSlug . '/?#';
-				$replace = '/' . $this->listSlug . '/';
-				$redirect_url = preg_replace( $search, $replace, $_SERVER['REQUEST_URI'] );
-				$redirect_url = esc_url_raw( add_query_arg( array( 'tribe_event_display' => 'past' ), $redirect_url ) );
-				wp_redirect( $redirect_url );
-				die;
-			} elseif ( strpos( $_SERVER['REQUEST_URI'], $this->getRewriteSlug() . '/' . $this->upcomingSlug ) !== false ) {
-				$search = '#/' . $this->upcomingSlug . '/?#';
-				$replace = '/' . $this->listSlug . '/';
-				$redirect_url = preg_replace( $search, $replace, $_SERVER['REQUEST_URI'] );
-				wp_redirect( $redirect_url );
-				die;
-			}
-
+		    // We are only interested in the path and not any query args
+            if ( ! $requested_path = parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH ) ) {
+                return;
+            }
+
+            // Prep strings so we can compare paths sans any leading/trailing slashes
+            $requested_path = trim( $requested_path, '/' );
+            $legacy_past_events_url = $this->getRewriteSlug() . '/' . $this->pastSlug;
+			$legacy_upcoming_events_url = $this->getRewriteSlug() . '/' . $this->upcomingSlug;
+
+			// Check if the request is for either the legacy past or upcoming event views and redirect if appropriate
+            switch ( $requested_path ) {
+                case $legacy_past_events_url:
+	                $search = '#/' . $this->pastSlug . '/?#';
+	                $replace = '/' . $this->listSlug . '/';
+	                $redirect_url = preg_replace( $search, $replace, $_SERVER['REQUEST_URI'] );
+	                $redirect_url = esc_url_raw( add_query_arg( array( 'tribe_event_display' => 'past' ), $redirect_url ) );
+	                wp_redirect( $redirect_url );
+	                exit;
+
+                case $legacy_upcoming_events_url:
+	                $search = '#/' . $this->upcomingSlug . '/?#';
+	                $replace = '/' . $this->listSlug . '/';
+	                $redirect_url = preg_replace( $search, $replace, $_SERVER['REQUEST_URI'] );
+	                wp_redirect( $redirect_url );
+	                exit;
+            }
```

Comment:
Weird indentation. I would move the `wp_redirect()` and the `exit` to outside of the switch.
And add a Default condition to the Switch.

Reply:
Also could use `tribe_exit()`.

Reply:
Will fix indentation and switch to using tribe_exit() ... I don't want to redirect or exit if those two conditions are not met, however.